fbw-a380x/src/systems/systems-host/systems/LegacyFuel.ts

```diff
@@ -0,0 +1,431 @@
+
+  private readonly junctionSettings = new Map<number, ConsumerSubject<number>>();
+
+  private refuelInProgress = false;
+
+  constructor(private readonly bus: EventBus) {
+    KeyEventManager.getManager(bus).then((manager) => {
+      this.keyEventManager = manager;
+    });
+
+    for (let index = 1; index <= this.NUMBER_OF_TRIGGERS; index++) {
+      const element = ConsumerSubject.create(this.sub.on(`fuel_trigger_status_${index}`), false);
+      this.triggerStates.set(index, element);
+    }
+
+    for (let index = 1; index <= this.NUMBER_OF_JUNCTIONS; index++) {
+      const element = ConsumerSubject.create(this.sub.on(`fuel_junction_setting_${index}`), 1);
+      this.junctionSettings.set(index, element);
+    }
+  }
+
+  init() {
+    const fuelWeight = SimVar.GetSimVarValue('FUEL TOTAL QUANTITY WEIGHT', 'kilograms');
+    SimVar.SetSimVarValue('L:A32NX_FUEL_DESIRED', 'kilograms', fuelWeight);
+    Wait.awaitSubscribable(GameStateProvider.get(), (state) => state === GameState.ingame, true).then(() => {
+      this.checkEmptyTriggers();
+    });
+  }
+
+  private checkEmptyTriggers(): void {
+    if (
+      (this.leftInnerTankQty.get() < 0.1 && !this.triggerStates.get(11).get()) ||
+      (this.leftInnerTankQty.get() >= 1 && this.triggerStates.get(11).get())
+    ) {
+      this.toggleTrigger(11);
+    }
+
+    if (
+      (this.rightInnerTankQty.get() < 0.1 && !this.triggerStates.get(12).get()) ||
+      (this.rightInnerTankQty.get() >= 1 && this.triggerStates.get(12).get())
+    ) {
+      this.toggleTrigger(12);
+    }
+
+    if (
+      (this.leftMidTankQty.get() < 0.1 && !this.triggerStates.get(22).get()) ||
+      (this.leftMidTankQty.get() >= 1 && this.triggerStates.get(22).get())
+    ) {
+      this.toggleTrigger(22);
+    }
+
+    if (
+      (this.rightMidTankQty.get() < 0.1 && !this.triggerStates.get(23).get()) ||
+      (this.rightMidTankQty.get() >= 1 && this.triggerStates.get(23).get())
+    ) {
+      this.toggleTrigger(23);
+    }
+
+    if (
+      this.rightInnerTankQty.get() < 0.1 &&
+      this.leftInnerTankQty.get() < 0.1 &&
+      this.rightMidTankQty.get() < 0.1 &&
+      this.leftMidTankQty.get() < 0.1
+    ) {
+      // both mid and inner tanks are empty
+
+      this.setJunctionOption(10, 3);
+    } else if (this.rightInnerTankQty.get() >= 1 || this.leftInnerTankQty.get() >= 1) {
+      // inner tanks arent empty
+
+      this.setJunctionOption(10, 1);
+    } else {
+      // mid tanks arent empty but inner tanks are
+
+      this.setJunctionOption(10, 2);
+    }
+
+    if (
+      (this.trimTankQty.get() < 0.1 && !this.triggerStates.get(32).get()) ||
+      (this.trimTankQty.get() >= 1 && this.triggerStates.get(32).get())
+    ) {
+      this.toggleTrigger(32);
+    }
+  }
+
+  public onUpdate(): void {
+    const onGround = SimVar.GetSimVarValue('SIM ON GROUND', 'bool');
+    if (this.refuelStarted.get()) {
+      this.refuelInProgress = true;
```

Comment:
NItpick: maybe call it something like `refuelWasInProgress`, had to think a bit what the reasoning of this field is

Reply:
Yes the idea was to notice when fuelling has ended. But the var itself really contains the information whether fueling is happening right now. As I turn it to false once I detect that fuelling isnt happening anymore. Hence I felt that this name is more appropriate.